core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -210,6 +212,31 @@ public Runtime build() {
         }
     }
 
+    private static final class CucumberThreadFactory implements ThreadFactory {
+
+        private final AtomicInteger threadNumber = new AtomicInteger(1);
+        private final ThreadGroup group;
+
+        CucumberThreadFactory() {
+            SecurityManager s = System.getSecurityManager();
+            this.group = s != null ? s.getThreadGroup() : Thread.currentThread().getThreadGroup();
+        }
+        
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = new Thread(this.group, r, "cucumber-runner-thread-" + this.threadNumber.getAndIncrement(), 0L);
```

Comment:
Using the `Thread(String name)` constructor should be the safest, least confusing option here.